Recruit-Training/Embedded-Recruit-Training/Core/Src/BlueLedToggleTask.c

```diff
@@ -1,16 +1,46 @@
 #include "BlueLedToggleTask.h"
+static const BLUE_LED_STATUS_STDID = 0xCCC;
+const ONE_HZ_TO_MS = 1000;
 
 void blueLedToggleTask(void const* arg)
 {
+    static const BLUE_LED_TOGGLE_FREQ = ONE_HZ_TO_MS;
     //One time osDelayUntil initialization
     uint32_t prevWakeTime = osKernelSysTick();
 
     osMutexId_t* canMutex = (osMutexId_t*)arg; // Get mutex that was passed as an argument
 
     for (;;)
     {
-        osDelayUntil(prevWakeTime + 1);  //TODO: Replace 1 with time period for delay
+        osDelayUntil(prevWakeTime + BLUE_LED_TOGGLE_FREQ);  //TODO: Replace 1 with time period for delay
         //TODO: Check blue toggle flag and toggle blue LED
+        if(blueFlag == 1)
+        {
+            HAL_GPIO_TogglePin(GPIOA, GPIO_PIN_5);
+        }
+        GPIO_PinState pinStatus = HAL_GPIO_ReadPin(GPIOA, GPIO_PIN_5);
         //TODO: Send CAN message indicating current state of LED
+        osStatus_t status = osMutexAcquire(canMutex, 1);
+        while(status != osOK){
```

Comment:
Instead of waiting for the status to be ```OK``` we usually just check the status and continue if ```!=osOK```
```suggestion
          if (osMutexAcquire(canMutex, 0) == osOK)
```